webapp/components/Image/index.vue

```diff
@@ -0,0 +1,23 @@
+<template>
+  <img
+    v-bind="imageProps"
+    :src="imageSrc"
+  >
+</template>
+
+<script>
+export default {
+  props: {
+    imageProps: {
+      type: Object,
+      required: true
+    }
+  },
+  computed: {
+    imageSrc() {
+      const src = this.imageProps.src
+      return src.startsWith('/') ? src.replace('/', '/api/') : src
```

Comment:
> [<img alt="tansaku" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/tansaku) **Authored by [tansaku](https://github.com/tansaku)**
_<time datetime="2019-05-02T13:28:41Z" title="Thursday, May 2nd 2019, 3:28:41 pm +02:00">May 2, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,23 @@
+<template>
+  <img
+    v-bind="imageProps"
+    :src="imageSrc"
+  >
+</template>
+
+<script>
+export default {
+  props: {
+    imageProps: {
+      type: Object,
+      required: true
+    }
+  },
+  computed: {
+    imageSrc() {
+      // TODO implement
+      return 'this should be an image source attribute'
```


so here we need something like:

```js
return  src.startsWith('/img') ? src.replace('/img','/api/img') : src
```
assuming that we can refer to the current `src` value here

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-02T13:59:31Z" title="Thursday, May 2nd 2019, 3:59:31 pm +02:00">May 2, 2019</time>_

---


No, check if it starts with `/`. I believe NodeJS URL module can tell you if it's absolute or not. For the browser, there should be a polyfill. See if it's possible: https://nodejs.org/api/url.html
If not, feel free to assume that a link starting with `/` is relative.

Reply:
> [<img alt="tansaku" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/tansaku) **Authored by [tansaku](https://github.com/tansaku)**
_<time datetime="2019-05-02T15:19:30Z" title="Thursday, May 2nd 2019, 5:19:30 pm +02:00">May 2, 2019</time>_

---


@roschaefer but isn't it the case that we only want to rewrite img links?  or you're saying we'd like to prefix *every* relative url with `/api`?  if so fair enough :-)

It doesn't look like NodeJS URL module has a check for that https://nodejs.org/api/url.html some good discussion of alternatives here: https://stackoverflow.com/questions/10687099/how-to-test-if-a-url-string-is-absolute-or-relative  

looks like [urijs](http://medialize.github.io/URI.js/docs.html#is) provides the functionality you're suggesting, and that's not in the project yet (although could be a good add), and that would give us code like:

```js
var uri = new URI(src);
return uri.is("relative") ? src.replace('^/','/api/') : src
```
Although I can see potential issues there if there are links that are relative to the current path, e.g. "../foo/bar", although perhaps that won't ever occur?

@aonomike hope you're following all this :-)

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-02T15:46:51Z" title="Thursday, May 2nd 2019, 5:46:51 pm +02:00">May 2, 2019</time>_

---


@tansaku the component we're talking about is called `ImageComponent` so a developer who would use this component will use it only for images (hopefully :disappointed: ). We can assume that we have *all* our images at some `uploads` service (currently the backend serving `public`).

You *could* add another npm package to check absolute/relative but I think this is quite an overkill.  

@tansaku it seems like you understand the problem but don't want to provide the fix yourself, is that correct? Were you successful setting up the project? @aonomike is not that responsive these days so I would welcome if you could provide the commit yourself.

Issue #296 was created on February 5th and if this conversation takes any longer I'll probably take care of it myself.

Reply:
> [<img alt="tansaku" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/tansaku) **Authored by [tansaku](https://github.com/tansaku)**
_<time datetime="2019-05-03T07:57:59Z" title="Friday, May 3rd 2019, 9:57:59 am +02:00">May 3, 2019</time>_

---


@roschaefer based on your great input I could now easily provide the fix - as you have inferred I'm holding off because I'm trying to provide a learning opportunity for @aonomike (and/or others) - hopefully I'll see him in our AV scrum today and we can take care of it together.